base/linalg/blas.jl

```diff
@@ -217,20 +217,41 @@ scal(n, DA, DX, incx) = scal!(n, DA, copy(DX), incx)
 
 Dot product of two vectors consisting of `n` elements of array `X` with stride `incx` and
 `n` elements of array `Y` with stride `incy`.
+
+**Example**:
```

Comment:
The doc style guide suggests using `# Examples`, though I don't think it's being followed that widely in base yet; perhaps due to how headers are currently rendered in the REPL (they take up a bit to much vertical space I think).